fdbserver/workloads/HealthMetricsApi.actor.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * WriteThrottling.actor.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+//
+// Created by sfc-gh-xwang on 6/30/20.
+//
+
+#include "fdbserver/TesterInterface.actor.h"
+#include "fdbserver/workloads/workloads.actor.h"
+#include "fdbserver/WorkerInterface.actor.h"
+#include "flow/actorcompiler.h" // This must be the last #include.
+
+// workload description
+// This workload can be attached to other workload to collect health information about the FDB cluster.
+struct HealthMetricsApiWorkload : TestWorkload {
+    // Performance Metrics
+    int64_t worstStorageQueue;
+    int64_t worstStorageDurabilityLag;
+    int64_t worstTLogQueue;
+    int64_t detailedWorstStorageQueue;
+    int64_t detailedWorstStorageDurabilityLag;
+    int64_t detailedWorstTLogQueue;
+    double detailedWorstCpuUsage;
+    double detailedWorstDiskUsage;
+
+    // Test configuration
+    double testDuration;
+    double healthMetricsCheckInterval;
+    double maxAllowedStaleness;
+    bool sendDetailedHealthMetrics;
+
+    // internal states
+    bool healthMetricsStoppedUpdating;
+
+    HealthMetricsApiWorkload(WorkloadContext const& wcx) : TestWorkload(wcx) {
+        testDuration = getOption(options, LiteralStringRef("testDuration"), 120.0);
+        healthMetricsCheckInterval = getOption(options, LiteralStringRef("healthMetricsCheckInterval"), 1.0);
+        sendDetailedHealthMetrics = getOption(options, LiteralStringRef("sendDetailedHealthMetrics"), true);
+        maxAllowedStaleness = getOption(options, LiteralStringRef("maxAllowedStaleness"), 10.0);
+    }
+
+    virtual std::string description() { return "HealthMetricsApiWorkload"; }
```

Comment:
```suggestion
    static constexpr const char* NAME = "HealthMetricsApiWorkload";
    virtual std::string description() { return NAME; }
```

This way, if someone ever want to change the name of the workload they only need to change it in one place.

Reply:
This is a good suggestion on the existing workloads as well. Not sure if @sfc-gh-xwang will be willing to help clean it up in a different PR.. :)

Reply:
Do you mean all other workloads we already have?
I can do that while reading other workloads, assuming this is not that urgent

Reply:
Yes. It is not urgent. It is mostly mechanical work.